### PR TITLE
ci: refactor src paths, remove archive, update stats, replace lodash with ramda in core

### PR DESCRIPTION
**Pull request by bardisg**
## Description of the change

Main change is the additing of github actions to automate our pipeline.

To support this, included changes are:

- replace core js lodash with ramda equivalents
- clean folder structure
- remove archive folders
- update stats

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

[> Fix [#1]() ](https://www.notion.so/rudderstacks/CI-CD-pipeline-JS-3d99ddf9b6f64501acb0cd567ced65ed)

### Development

- [ ] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [x] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/724)

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,5 +1,8 @@
...
-/**/*.d.ts
-/node_modules/
-/dist/**/*.js
+dist/
+reports/
+.husky/
+__mocks__/
+*.json
+*.md
+*.d.ts
```
Comment (saikumarrs) [inline .eslintignore:7]: We create this declaration file manually. Hence, it's better to have it.

File: .eslintignore [hunk 2]
```diff
@@ -1,5 +1,8 @@
-**/*test.js
-/**/*.d.ts
-/node_modules/
```
Comment (saikumarrs) [inline .eslintignore:3]: We should ignore node_modules.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+- Chrome
+  >
+- Firefox
+  >
+- Safari
+  >
+
+## Security
+
+- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.
```
Comment (saikumarrs) [inline .github/pull_request_template.md:22]: These are not strictly applicable to the SDK. I'd rather tweak the existing org template.

**Review by saikumarrs — COMMENTED** [review]
Partial review 1.

File: .github/workflows/deploy-dev.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+      - name: Build files
+        env:
+          HUSKY: 0
+        run: |
+          npm ci
+          npm run build:browser:dev
+          npm run build:integration:all
+
+      - name: Fix BugSnug API key
```
Comment (saikumarrs) [inline .github/workflows/deploy-dev.yml:44]:
```suggestion
      - name: Fix Bugsnag API key
```

File: .github/workflows/deploy-npm.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Checkout source branch
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+
+      - name: Build fils
```
Comment (saikumarrs) [inline .github/workflows/deploy-npm.yml:25]:
```suggestion
      - name: Build files
```

**Review by saikumarrs — COMMENTED** [review]
Partial review 2

File: .gitignore [hunk 6]
```diff
@@ -1,24 +1,35 @@
...
 *.iml
-**/node_modules
-**/**/node_modules
 local.properties
 *.apk
 *.zip
 build.sh
 publish.sh
-dist/*
-!dist/rudder-sdk-js/*
```
[the remark this answers is not in the document]
Comment (bardisg) [reply .gitignore:18]: not anymore, the files come from packages/npm folder now and are never hardcoded in dist folder

File: .eslintrc.json [hunk 7]
```diff
@@ -1,14 +1,40 @@
...
   "parserOptions": {
     "ecmaVersion": 11,
     "sourceType": "module"
   },
-  "rules": { "import/prefer-default-export": "off" }
+  "rules": {
+    "import/prefer-default-export": "off",
+    "unicorn/filename-case": [
+      "error",
+      { "cases": { "camelCase": true, "pascalCase": true, "kebabCase": true } }
```
[the remark this answers is not in the document]
Comment (bardisg) [reply .eslintrc.json:19]: We use the same rules as control plane team to align organization wide

File: .eslintignore (hunk 1 above)
Comment (bardisg) [reply .eslintignore:7]: we cannot lint ts files without typescript configuration, lets add this is when we start moving to TS

File: .eslintignore (hunk 2 above)
Comment (bardisg) [reply .eslintignore:3]: will add this one

File: .github/pull_request_template.md (hunk 3 above)
Comment (bardisg) [reply .github/pull_request_template.md:22]: lets set a meeting this week with  @MoumitaM and decide on what we want to place in PR template

File: CHANGELOG.md [hunk 8]
```diff
@@ -44,7 +44,7 @@ Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).
 - fix(ga): adding support of revenue along with total. [`#690`](https://github.com/rudderlabs/rudder-sdk-js/pull/690)
 - feat: posthog group call support [`#688`](https://github.com/rudderlabs/rudder-sdk-js/pull/688)
 - chore(deps): bump @xmldom/xmldom from 0.7.5 to 0.7.6 [`#691`](https://github.com/rudderlabs/rudder-sdk-js/pull/691)
-- feat(new integration): refiner onboarding  [`#682`](https://github.com/rudderlabs/rudder-sdk-js/pull/682)
+- feat(new integration): refiner onboarding [`#682`](https://github.com/rudderlabs/rudder-sdk-js/pull/682)
```
[the remark this answers is not in the document]
Comment (bardisg) [reply CHANGELOG.md:47]: sure, lets do that to avoid any inconsistencies

File: README.md [hunk 9]
```diff
@@ -183,9 +183,9 @@ Refer to the following projects for a detailed walk-through of the above steps:
 ### Supported browser versions
 
 | **Browser**     | **Supported Versions** |
-| :-------------- | :--------------------- |
+| :-------------- |:-----------------------|
 | Safari          | v7 or later            |
-| IE              | v10 or later           |
+| IE              | v11 or later           |
```
[the remark this answers is not in the document]
Comment (bardisg) [reply README.md:188]: puting this back to 11, lets treat IE10 support drop as separate ticket

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -14,7 +14,7 @@ Thanks for taking the time and for your help in improving this project!
 
 ## RudderStack Contributor Agreement
 
-To contribute to this project, we need you to sign the [**Contributor License Agreement (“CLA”)**][CLA] for the first commit you make. By agreeing to the [**CLA**][CLA]
+To contribute to this project, we need you to sign the [**Contributor License Agreement (“CLA”)**][cla] for the first commit you make. By agreeing to the [**CLA**][cla]
```
[the remark this answers is not in the document]
Comment (bardisg) [reply CONTRIBUTING.md:17]: done, also fixed the relevant docs in the file

File: .github/workflows/deploy-dev.yml (hunk 4 above)
Comment (bardisg) [reply .github/workflows/deploy-dev.yml:44]: done

File: .github/workflows/deploy-npm.yml (hunk 5 above)
Comment (bardisg) [reply .github/workflows/deploy-npm.yml:25]: done

**Review by MoumitaM — APPROVED** [review]

File: CHANGELOG.md [hunk 11]
```diff
@@ -61,7 +61,7 @@ Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).
 - fix(ga): adding support of revenue along with total. [`#690`](https://github.com/rudderlabs/rudder-sdk-js/pull/690)
```
Comment (saikumarrs) [inline CHANGELOG.md:61]: Let's revert the changelog updates.

**Review by saikumarrs — DISMISSED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**